classes/ActionScheduler_ListTable.php

```diff
@@ -0,0 +1,385 @@
+			'jquery',
+			'jquery-ui-core',
+			'jquery-ui-dialog',
+		), '1.0', true );
+	}
+
+	/**
+	 * Returns the recurrence of an action or 'Non-repeating'. The output is human readable.
+	 *
+	 * @param ActionScheduler_Action $item
+	 *
+	 * @return string
+	 */
+	protected function get_recurrence( $item ) {
+		$recurrence = $item->get_schedule();
+		if ( method_exists( $recurrence, 'interval_in_seconds' ) ) {
+			return self::human_interval( $recurrence->interval_in_seconds() );
+		}
+		return __( 'Non-repeating', 'action-scheduler' );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function get_items_query_limit() {
+		return $this->get_items_per_page( $this->package . '_items_per_page', $this->items_per_page );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function get_items_query_offset() {
+		global $wpdb;
+
+		$per_page = $this->get_items_query_limit();
+		$current_page = $this->get_pagenum();
+		if ( 1 < $current_page ) {
+			return $per_page * ( $current_page - 1 );
+		}
+
+		return 0;
+	}
+
+	/**
+	 * Serializes the argument of an action to render it in a human friendly format.
+	 *
+	 * @param array $row The array representation of the current row of the table
+	 *
+	 * @return string
+	 */
+	public function column_args( array $row ) {
+		return '<code>' . json_encode( $row['args'] ) . '</code>';
+	}
+
+	/**
+	 * Prints the comments, which are the log entries. It needs to be named comments otherwise it won't pickup
+	 * the WordPress styles to make the number pretty.
+	 *
+	 * It will also render all the log entries, but it will be hidden and rendered on click in the number as a modal.
+	 *
+	 * @param array $row Action array.
+	 */
+	public function column_comments( array $row ) {
+		echo '<div id="log-' . $row['ID'] . '" class="log-modal hidden" style="max-width:800px">';
+		echo '<h3>' . esc_html( sprintf( __( 'Log entries for %d', 'action-scheduler' ),  $row['ID'] ) ) . '</h3>';
+		$timezone = new DateTimezone( 'UTC' );
+		foreach ( $row['comments'] as $log ) {
+			$date = $log->get_date();
+			$date->setTimezone( $timezone );
+			echo '<p><strong>' . esc_html( $date->format( 'Y-m-d H:i:s' ) ) . ' UTC</strong> ' . esc_html( $log->get_message() ) . '</p>';
+		}
+		echo '</div>';
+
+		echo '<div class="post-com-count-wrapper">';
+		echo '<a href="#" data-log-id="' . $row['ID'] . '" class="post-com-count post-com-count-approved log-modal-open">';
+		echo '<span class="comment-count-approved">' . esc_html( count( $row['comments'] ) ) . '</span>';
+		echo '</a>';
+		echo '</div>';
+	}
+
+	/**
+	 * Prints the scheduled date in a human friendly format.
+	 *
+	 * @param array $row The array representation of the current row of the table
+	 */
+	public function column_scheduled( $row ) {
+		$next_timestamp = $row['scheduled']->next()->format( 'U' );
+		echo get_date_from_gmt( gmdate( 'Y-m-d H:i:s', $next_timestamp ), 'Y-m-d H:i:s' );
```

Comment:
Why use `get_date_from_gmt()` and `gmdate()` here instead of just `$row['scheduled']->next()->format( 'Y-m-d H:i:s' )`?

The value should probably also be escaped before being printed just to be safe.